### PR TITLE
Add About dialog

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/deltachat/deltachat-desktop/issues/103

Will leave this hanging for someone else to tweak the dialog. I'm not skilled enough and/or don't have enough creativity to make something out of the dialog.

Currently my thoughts are:

* add some tabs
* default tab shows basic and clean About information (desktop version, license, yada yada)
* click the second tab to get more information

Also see old PR https://github.com/deltachat/deltachat-desktop/pull/197

### CODE REVIEW

File: src/main/deltachat.js [hunk 1]
```diff
@@ -145,6 +145,17 @@ class DeltaChatController {
     if (typeof this._render === 'function') this._render()
   }
 
+  /**
+   * Dispatched when showing the AboutDialog (owned by ScreenController)
```
Comment (ralphtheninja) [inline src/main/deltachat.js]: It has to be `ScreenController` that owns this, if we want to show it without being logged in.

File: src/main/deltachat.js [hunk 2]
```diff
@@ -145,6 +145,17 @@ class DeltaChatController {
...
+   */
+  getInfo () {
+    if (this.ready === true) {
+      return this._dc.getInfo()
+    } else {
+      return DeltaChat.getSystemInfo()
```
Comment (ralphtheninja) [inline src/main/deltachat.js]: This returns basic system information, which is a stripped down version of `dc.getInfo()`

File: src/renderer/ScreenController.js [hunk 3]
```diff
@@ -37,13 +44,27 @@ class Home extends React.Component {
...
+        'dispatchSyncNoRender',
+        'getInfo'
+      )
+    }
+    this.setState({ showAbout, aboutInfo })
   }
 
   render () {
-    // renderer/main.js polls every second and updates the deltachat
-    // property with current state of database.
```
Comment (ralphtheninja) [inline src/renderer/ScreenController.js]: Old comment, we're no longer polling for data.

File: src/main/ipc.js [hunk 4]
```diff
@@ -54,6 +54,10 @@ function init (cwd) {
     e.returnValue = dispatch(...args)
   })
 
+  ipc.on('dispatchSyncNoRender', (e, ...args) => {
+    e.returnValue = dispatch(...args)
+  })
```
Comment (ralphtheninja) [inline src/main/ipc.js]: This should be removed now.

File: src/main/deltachat.js (hunk 1 above)
Comment (karissa) [reply src/main/deltachat.js]: Yeah I have an idea on how to refactor this with a sort of `DialogController`, can take a look.

**Comment by ralphtheninja** [comment]
@Jikstra Have you tried opening the dialog and display the advanced stuff while logged in? I'm thinking maybe we should either limit the info or somehow tweak the ui so we can display long strings in a nicer way, e.g. via line breaks etc

**Comment by Jikstra** [comment]
> @Jikstra Have you tried opening the dialog and display the advanced stuff while logged in? I'm thinking maybe we should either limit the info or somehow tweak the ui so we can display long strings in a nicer way, e.g. via line breaks etc

Yes did it. It's ugly but not really possible to do it differently with the current html. We would need a table or something. I didn't want to further work on it as i think it's not the right place to put those infos. It's just a hack until we have settings in my opinion.

File: src/main/deltachat.js (hunk 2 above)
Comment (ralphtheninja) [reply src/main/deltachat.js]: @Jikstra We could skip the top line and just do `return DeltaChat.getSystemInfo()`.

**Comment by ralphtheninja** [comment]
I think we should do one out of the following two things to round this off:

* _remove_ the collapse thingie entirely and just display a simple/clean about dialog (and move this info elsewhere in another issue)
* _only_ do `DeltaChat.getSystemInfo()` (this will give more details about the software, which I think is nice and at the same time don't be _too_ ugly with enormously long strings)

**Comment by Jikstra** [comment]
> remove the collapse thingie entirely and just display a simple/clean about dialog (and move this info elsewhere in another issue)

I'm pro this. Should be somewhere in Settings->Advanced->Debug Information or something

**Comment by karissa** [comment]
Yeah I think removing the collapse thingy is a good idea. Removing complexity

**Comment by ralphtheninja** [comment]
Anyone against merging this now?

**Comment by Jikstra** [comment]
> Anyone against merging this now?

Nope. LGTM let's merge.

File: src/renderer/ScreenController.js [hunk 5]
```diff
@@ -87,6 +100,7 @@ class Home extends React.Component {
             deltachat={deltachat}
           />
         }
+        { showAbout && (<AboutDialog isOpen={showAbout} onClose={this.onCloseAbout} />) }
```
Comment (karissa) [inline src/renderer/ScreenController.js]: you don't need showAbout && if you have an `isOpen` parameter.

File: src/renderer/components/dialogs/About.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    return (
+      <Dialog
+        isOpen={isOpen}
+        title={tx('aboutDeltaChat')}
+        icon='info-sign'
+        onClose={onClose}
+        canOutsideClickClose={false}>
+        <div className={Classes.DIALOG_BODY}>
+          <p style={{ color: 'grey' }}>{'version ' + version}</p>
+          <p>Official Delta Chat Desktop app.</p>
```
Comment (karissa) [inline src/renderer/components/dialogs/About.js]: we'll have to translate this but I guess it's secondary

File: src/renderer/ScreenController.js (hunk 5 above)
Comment (ralphtheninja) [reply src/renderer/ScreenController.js]: More specifically, the overlay div will be part of the dom once it has been rendered:

screenshot from 2018-11-01 10-47-55

Lets just remove it. Makes the code cleaner as well.

Comment (ralphtheninja) [reply src/renderer/ScreenController.js]: https://github.com/deltachat/deltachat-desktop/commit/a8a4710f15f94c8330dac5d815b51617bd6294e4

**Pull request merged**